src/coreComponents/dataRepository/Wrapper.hpp

```diff
@@ -276,80 +276,163 @@ class Wrapper final : public WrapperBase
 
   ///////////////////////////////////////////////////////////////////////////////////////////////////
   /// @copydoc WrapperBase::isPackable
-  virtual
-  bool isPackable( bool onDevice ) const override
+  template < typename U = T >
```

Comment:
Do you only want `U == T`? Is there a way to assert it?

Reply:
I'll make all the pack `impl` functions `private`, since the intent is only ever to access them through the packing interfaces, and those instantiations will only ever be `T`. 

We need to have this `U=T` template since SFINAE only works when we're using template type deduction, which if we directly use `T` which is known inside the class instantiation, we're not.

Reply:
Could we `static_assert` + `is_same` something or that would not be working?

Reply:
I think as an additional layer of security to prevent internal misuse I can add a static assertion, yeah.